arm-recoveryservices/2015-01-01/swagger/recoveryservices.json

```diff
@@ -0,0 +1,285 @@
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/ResourceOutput"
+            }
+          },
+          "201": {
+            "description": "Created",
+            "schema": {
+              "$ref": "#/definitions/ResourceOutput"
+            }
+          }
+        },
+        "deprecated": false
+      },
+      "delete": {
+        "tags": [
+          "Vaults"
+        ],
+        "description": "Deletes a vault.",
+        "operationId": "Vaults_Delete",
+        "produces": [
+          "application/json",
+          "text/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionId"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupName"
+          },
+          {
+            "$ref": "#/parameters/VaultName"
+          },
+          {
+            "name": "isGlobalRPContext",
+            "in": "query",
+            "required": false,
+            "type": "boolean"
+          }
+        ],
+        "responses": {
+          "204": {
+            "description": "NoContent"
+          }
+        },
+        "deprecated": false
+      }
+    }
+  },
+  "definitions": {
+    "Object": {
+      "description": "Base of all objects.",
+      "type": "object",
+      "properties": {}
+    },
+    "ResourceCollectionOutput": {
+      "description": "The response model for Vault.",
+      "type": "object",
+      "properties": {
+        "Value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ResourceOutput"
+          }
+        }
+      }
+    },
+    "ResourceOutput": {
+      "description": "Resource information, as returned by the resource provider.",
+      "type": "object",
+      "properties": {
+        "location": {
+          "description": "Location of the resource.",
+          "type": "string"
+        },
+        "name": {
+          "description": "The name of the resource.",
+          "type": "string"
+        },
+        "etag": {
+          "description": "ETag of the Vault.",
+          "type": "string"
+        },
+        "tags": {
+          "description": "The tags attached to the resource.",
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        },
+        "properties": {
+          "$ref": "#/definitions/ResourceProperties"
+        },
+        "id": {
+          "description": "The Id of the resource.",
+          "type": "string"
+        },
+        "type": {
+          "description": "Type of the resource.",
+          "type": "string"
+        }
+      }
+    },
+    "Sku": {
+      "description": "Identifies the unique system identifier for each Azure resource.",
+      "type": "object",
+      "properties": {
+        "name": {
+          "description": "Name.",
+          "type": "string"
+        }
+      }
+    },
+    "ResourceInput": {
```

Comment:
ResourceInput and ResourceOutput can be merged into one object.
Basically your body parameter for PUT, response of PUT, response of GET all can be same the model type "Vault", if you model it correctly. This will make it easier in piping scenario.

That is you create a model called "Vault" once and use it as input for create, use the same model for response of create.

use "Vault" for GET; make changes to its updatable properties and then send it as input to your PUT (update) request. 

Having different models for GET and PUT causes customers to copy the model properties and that is tedious.